parse/ast.go

```diff
@@ -347,6 +343,8 @@ func (ql QuotedLiteral) String() string {
 
 	return fmt.Sprintf("|%s|", r.Replace(string(ql)))
 }
+func (nl NameLiteral) String() string   { return string(nl) }
+func (nl NumberLiteral) String() string { return fmt.Sprint(float64(nl)) }
```

Comment:
Why do you need additional `float64()` cast?

Reply:
without casting infinite recursion :)

Reply:
as fmt.Sprint(nl), will call nl.String(), which will cal fmt.Sprint(nl), and so on and on

Reply:
Indeed.